ql/src/java/org/apache/hadoop/hive/ql/exec/tez/KillTriggerActionHandler.java

```diff
@@ -38,28 +38,24 @@ public class KillTriggerActionHandler implements TriggerActionHandler<TezSession
   @Override
   public void applyAction(final Map<TezSessionState, Trigger> queriesViolated) {
     for (Map.Entry<TezSessionState, Trigger> entry : queriesViolated.entrySet()) {
-      switch (entry.getValue().getAction().getType()) {
-      case KILL_QUERY:
-        TezSessionState sessionState = entry.getKey();
-        String queryId = sessionState.getWmContext().getQueryId();
-        try {
-          UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
-          SessionState ss = new SessionState(new HiveConf(), ugi.getShortUserName());
-          ss.setIsHiveServerQuery(true);
-          SessionState.start(ss);
-          KillQuery killQuery = sessionState.getKillQuery();
-          // if kill query is null then session might have been released to pool or closed already
-          if (killQuery != null) {
-            sessionState.getKillQuery().killQuery(queryId, entry.getValue().getViolationMsg(),
-                      sessionState.getConf());
-          }
-        } catch (HiveException|IOException e) {
-          LOG.warn("Unable to kill query {} for trigger violation", queryId);
+        if (entry.getValue().getAction().getType() == Action.Type.KILL_QUERY) {
+            TezSessionState sessionState = entry.getKey();
+            String queryId = sessionState.getWmContext().getQueryId();
+            try {
+                UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
+                DriverUtils.setUpSessionState(sessionState.getConf(), ugi.getShortUserName());
```

Comment:
Should we pass the `conf` to KillTriggerActionHandler constructor from TezSessionPoolManager?

Reply:
Before, when we created a new session, we passed a `new HiveConf`, but now that might overwrite the current session conf. Note that in `WorkloadManager` you are using global conf and not the `TezSessionState`

Reply:
What about `WorkloadManager`? does it use a separate thread to do it's job as well?